internal/api/grpc/admin/limits_integration_test.go

```diff
@@ -0,0 +1,99 @@
+//go:build integration
+
+package admin_test
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/zitadel/zitadel/pkg/grpc/admin"
+)
+
+func TestServer_Limits_DisallowPublicOrgRegistration(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+	domain, _, iamOwnerCtx := Tester.UseIsolatedInstance(ctx, SystemCTX)
+	regOrgUrl, err := url.Parse("http://" + domain + ":8080/ui/login/register/org")
+	require.NoError(t, err)
+	// The CSRF cookie must be sent with every request.
+	// We can simulate a browser session using a cookie jar.
+	jar, err := cookiejar.New(nil)
+	require.NoError(t, err)
+	browserSession := &http.Client{Jar: jar}
+	// Default should be allowed
+	csrfToken := awaitAllowed(t, iamOwnerCtx, browserSession, regOrgUrl)
+	_, err = Tester.Client.Admin.SetInstanceLimits(iamOwnerCtx, &admin.SetInstanceLimitsRequest{DisallowPublicOrgRegistration: true})
+	require.NoError(t, err)
+	awaitDisallowed(t, iamOwnerCtx, browserSession, regOrgUrl, csrfToken)
+	_, err = Tester.Client.Admin.ResetInstanceLimits(iamOwnerCtx, &admin.ResetInstanceLimitsRequest{})
+	require.NoError(t, err)
+	awaitAllowed(t, iamOwnerCtx, browserSession, regOrgUrl)
+}
+
+// awaitAllowed doesn't accept a CSRF token, as we expected it to always produce a new one
+func awaitAllowed(t *testing.T, ctx context.Context, client *http.Client, parsedURL *url.URL) string {
+	csrfToken := awaitGetResponse(t, ctx, client, parsedURL, http.StatusOK)
+	awaitPostFormResponse(t, ctx, client, parsedURL, http.StatusOK, csrfToken)
+	return csrfToken
+}
+
+// awaitDisallowed accepts an old CSRF token, as we don't expect to get a CSRF token from the GET request anymore
+func awaitDisallowed(t *testing.T, ctx context.Context, client *http.Client, parsedURL *url.URL, reuseOldCSRFToken string) {
+	awaitGetResponse(t, ctx, client, parsedURL, http.StatusNotFound)
+	awaitPostFormResponse(t, ctx, client, parsedURL, http.StatusConflict, reuseOldCSRFToken)
+}
+
+// awaitGetResponse cuts the CSRF token from the response body if it exists
+func awaitGetResponse(t *testing.T, ctx context.Context, client *http.Client, parsedURL *url.URL, expectCode int) string {
+	var csrfToken []byte
+	await(t, ctx, func() bool {
+		resp, err := client.Get(parsedURL.String())
+		require.NoError(t, err)
+		body, err := io.ReadAll(resp.Body)
+		require.NoError(t, err)
+		searchField := `<input type="hidden" name="gorilla.csrf.Token" value="`
+		_, after, hasCsrfToken := bytes.Cut(body, []byte(searchField))
+		if hasCsrfToken {
+			csrfToken, _, _ = bytes.Cut(after, []byte(`">`))
+		}
```

Comment:
I know this is rather ugly :smiling_face_with_tear: 
But in order to test the POST response, we need to have the execution go into the handler.
That's why we need to extract the CSRF token.
Otherwise the CSRF interceptor doesn't let the POST request through.